src/javascript/app_2/Modules/Trading/Containers/purchase.jsx

```diff
@@ -46,15 +49,34 @@ const Purchase = ({
                             currency={currency}
                             proposal_info={info}
                         />
-                        <Button
-                            is_disabled={is_disabled}
-                            id={`purchase_${type}`}
-                            className='primary green'
-                            has_effect
-                            text={localize('Purchase')}
-                            onClick={() => { onClickPurchase(info.id, info.stake); }}
-                            wrapperClassName='submit-section'
-                        />
+                        {is_purchase_confirm_on ?
+                            <PopConfirm
+                                alignment='left'
+                                cancel_text='Cancel'
+                                confirm_text='Purchase'
+                                message='Are you sure you want to purchase this contract?'
+                            >
+                                <Button
+                                    is_disabled={is_disabled}
+                                    id={`purchase_${type}`}
+                                    className='primary green'
+                                    has_effect
+                                    text={localize('Purchase')}
+                                    onClick={() => { onClickPurchase(info.id, info.stake); }}
+                                    wrapperClassName='submit-section'
+                                />
+                            </PopConfirm>
+                            :
+                            <Button
+                                is_disabled={is_disabled}
+                                id={`purchase_${type}`}
+                                className='primary green'
+                                has_effect
+                                text={localize('Purchase')}
+                                onClick={() => { onClickPurchase(info.id, info.stake); }}
+                                wrapperClassName='submit-section'
+                            />
+                        }
```

Comment:
`<Button />` in line 59 and 70 are the same, suggestion to declare the Button component maybe in line 29 and use that instead :)